### PR TITLE
NF-451 Info Line for lookup page

### PR DESCRIPTION
**Pull request by njmoorewitchcat**

### CODE REVIEW

**Comment by njmoorewitchcat** [comment]
Before and Default Behavour, when afterHeadingText is not supplied: 
BeforeAndDefault

After change - when afterHeadingText is supplied: 
AfterChange

File: README.md [hunk 1]
```diff
@@ -127,6 +127,7 @@ If your service doesn't have Welsh translations you can disable them setting the
       "lookupPageLabels": {
         "title": "Find address",
         "heading": "Find address",
+        "afterHeadingText": "foo",
```
Comment (SteveSugden) [inline README.md:130]: Replace with better description?

File: README.md [hunk 2]
```diff
@@ -173,6 +174,7 @@ If your service doesn't have Welsh translations you can disable them setting the
       "lookupPageLabels": {
         "title": "Find address welsh",
         "heading": "Find address welsh",
+        "afterHeadingText": "foo",
```
Comment (SteveSugden) [inline README.md:177]: Same

File: it/controllers/LookupPageISpec.scala [hunk 3]
```diff
@@ -169,6 +171,10 @@ class LookupPageISpec extends IntegrationSpecBase {
           doc.title shouldBe lookupPage.title.get + " - NAV_TITLE - GOV.UK"
           doc.h1.text() shouldBe lookupPage.heading.get
 
+          doc.getElementsByClass("afterHeadingText") should have(
```
Comment (SteveSugden) [inline it/controllers/LookupPageISpec.scala:174]: Should that be getElementById?

File: app/views/v2/lookup.scala.html [hunk 4]
```diff
@@ -45,6 +45,10 @@ <h1 class="govuk-heading-xl" id="pageHeading">
         @{if(isUKMode) messages("lookupPage.heading.ukMode") else messages("lookupPage.heading")}
       </h1>
 
+      @if(messages.isDefinedAt("lookupPage.afterHeadingText")) {
```
Comment (SteveSugden) [inline app/views/v2/lookup.scala.html:48]: Interesting.  What not pass an `afterHeadingMessage: Option[String]` parameter into the view (e.g. like `message`)?   Or is this the ALF way of doing "optional" content?

File: README.md (hunk 1 above)
Comment (njmoorewitchcat) [reply README.md:130]: It's an identical pattern to an optional text that is normally left blank.  See "headingWithPostcode" in the config section for selectPageLabels above.  The description for the field is below, and I think it's left this way so people don't copy/paste the config and it shows up clearly on the page as something you need to change.  Happy to change though, will get a proper steer from the ALF team.

File: it/controllers/LookupPageISpec.scala (hunk 3 above)
Comment (njmoorewitchcat) [reply it/controllers/LookupPageISpec.scala:174]: Yes it should - that would be intellij trying to be smart on me.

Comment (SteveSugden) [reply it/controllers/LookupPageISpec.scala:174]: There's a couple of others in this file I think

File: app/views/v2/lookup.scala.html (hunk 4 above)
Comment (rorticus0502) [reply app/views/v2/lookup.scala.html:48]: The message field is never used it was discussed with the team when we asked for permission to do this.
It appears to be a relic, something that was tried, never used or implemented fully.
There is no way to activate it from config.

But, removing here would be beyond the scope of what we agreed. They may well remove it.
HTH

Comment (njmoorewitchcat) [reply app/views/v2/lookup.scala.html:48]: I've taken it from the spike verbatium.  There is some magic that happens with messages where they get merged up from config.  This seems to be the less intrusive mechanism of showing or hiding this field if a key/value pair does not exist in the message bundle (after the RemoteMessageAPI does it's thing).  I will take a steer from ALF here though.

Comment (SteveSugden) [reply app/views/v2/lookup.scala.html:48]: My comment was more about the way we're conditionally rendering this new field - by checking for a definition in the messages file.  It's unusual, that's all.

**Pull request merged**